documentation/src/main/asciidoc/topics/proc_configuring_sql_stores_query.adoc

```diff
@@ -18,68 +28,61 @@ include::dependencies_maven/persistence_sql_store.xml[]
-+
-For example, specify if the cache store is shared between multiple cache instances with either the `shared` attribute or the `shared()` method.
+. Specify the database dialect with either `dialect=""` or `dialect()`, for example `dialect="H2"` or `dialect="postgres"`.
+. Configure the SQL cache store with the properties you require, for example:
 +
-If you want your SQL queries to update and delete entries, add the `read-only="false"` attribute or the `.ignoreModifications(false)` method to allow modifications to persistent storage.
+* To use the same cache store with multiple caches, set `shared="true"` or `shared(true)`.
 +
-. Add a JDBC connection factory so that {brandname} can connect to the database.
-. Specify the SQL query for loading data with either the `queries` element or the `queries()` method.
-.. Add `SELECT`, `SELECT ALL`, and `SIZE` clauses to your query.
-.. Add `DELETE`, `DELETE ALL`, and `UPSERT` clauses if the cache store allows modifications.
+* To create a read only cache store, set `read-only="true"` or `.ignoreModifications(true)`.
+
+. Define SQL query statements that load caches with data and modify database tables with the `queries` element or the `queries()` method.
 +
-* `SELECT`, `SELECT ALL` must return the same columns.
-* `DELETE` must take the same parameters as `SELECT`.
-* `UPSERT` must take the same named parameters as those returned by `SELECT` and `SELECT ALL`.
-. Configure a Protobuf schema if database tables contain composite keys or values.
-.. Add the `schema` element or the `.schemaJdbcConfigurationBuilder()` method and specify parameters for your Protobuf schema.
-.. Set the `embedded-key="true"` attribute or `.embeddedKey(true)` method if values also contain keys.
+[%autowidth,cols="1,1",stripes=even]
+|===
+|Query statement |Description
 
-[discrete]
-== SQL query store configuration
+|`SELECT`
+|Loads a single entry into caches.
+You can use wildcards but must specify parameters for keys.
+You can use labelled expressions.
 
-.XML
-[source,xml,options="nowrap",subs=attributes+,role="primary"]
-----
-include::xml/persistence_sql_query.xml[]
-----
+|`SELECT ALL`
+|Loads multiple entries into caches.
+You can use the `*` wildcard if the number of columns returned match the key and value columns.
+You can use labelled expressions.
 
-.JSON
-[source,json,options="nowrap",subs=attributes+,role="secondary"]
-----
-include::json/persistence_sql_query.json[]
-----
+|`SIZE`
+|Counts the number of entries in the cache.
 
-.YAML
-[source,yaml,options="nowrap",subs=attributes+,role="secondary"]
-----
-include::yaml/persistence_sql_query.yaml[]
-----
+|`DELETE`
+|Deletes a single entry from the cache.
 
-.ConfigurationBuilder
-[source,java,options="nowrap",subs=attributes+,role="secondary"]
-----
-include::code_examples/PersistenceSqlQuery.java[]
-----
+|`DELETE ALL`
+|Deletes all entries from the cache.
+
+|`UPSERT`
+|Modifies entries in the cache.
```

Comment:
The variables must have the same number of uniquely named variables that the select and select all return. For example if select returns `foo` and `bar` this statement must take only `:foo` and `:bar` as variables. Note that the same named variable can be applied more than once.